### PR TITLE
Merge 0.16.3

### PR DESCRIPTION
**Pull request by rossabaker**
There were several conflicts here, so doing this merge as a PR.  Travis' opinion is probably more useful than humans' on this one, but I'll listen to both.

### CODE REVIEW

File: project/Http4sPlugin.scala [hunk 1]
```diff
@@ -75,7 +75,7 @@ object Http4sPlugin extends AutoPlugin {
     http4sMimaVersion := {
       version.value match {
         case VersionNumber(Seq(major, minor, patch), _, _) if patch.toInt > 0 =>
-          Some(s"${major}.${minor}.0")
+          Some(s"${major}.${minor}.${patch.toInt - 1}")
```
Comment (rossabaker) [inline project/Http4sPlugin.scala]: 1. Scala 2.11 is stricter than 2.12, which we didn't discover until we ran on all branches.
2. What's done is done.

File: core/src/main/scala/org/http4s/util/io/package.scala [hunk 2]
```diff
@@ -83,7 +83,7 @@ package object io {
     captureOutputStream(osCallbackToWriterCallback(f))
 
   private def osCallbackToWriterCallback(f: JWriter => Unit) = { out: OutputStream =>
-    val w = new OutputStreamWriter(out)
+    val w = new OutputStreamWriter(out, StandardCharsets.UTF_8)
```
Comment (ChristopherDavenport) [inline core/src/main/scala/org/http4s/util/io/package.scala]: 😆 Fix into commented code.

File: project/Http4sPlugin.scala (hunk 1 above)
Comment (ChristopherDavenport) [reply project/Http4sPlugin.scala]: Not sure I follow on this one. We're only checking version to version not from initial binary for the minor?

File: core/src/main/scala/org/http4s/util/io/package.scala (hunk 2 above)
Comment (rossabaker) [reply core/src/main/scala/org/http4s/util/io/package.scala]: Well, if it ever becomes uncommented... :)

File: project/Http4sPlugin.scala (hunk 1 above)
Comment (rossabaker) [reply project/Http4sPlugin.scala]: Compatibility should be transitive.  If we haven't broken compatibility from .0, these should be equivalent.  If we have broken compatibility from .0, intentionally or not, then we have to choose between going back to .0 (another breaking change) or moving on with `${patch - 1}`.

**Comment by rossabaker** [comment]
Yes, most of this code is a rerun.  It was a non-trivial merge, so I decided to make a PR and solicit extra opinions.

**Pull request merged**